pkg/daemon/ceph/client/pool.go

```diff
@@ -496,17 +496,24 @@ func ensureFailureDomain(context *clusterd.Context, clusterInfo *ClusterInfo, cl
 	if err != nil {
 		return errors.Wrapf(err, "failed to get crush rule %q", details.CrushRule)
 	}
-	currentFailureDomain := extractFailureDomain(rule)
-	if currentFailureDomain == pool.FailureDomain {
-		logger.Debugf("pool %q has the expected failure domain %q", pool.Name, pool.FailureDomain)
+	currentFailureDomain, currentDeviceClass := extractPoolDetails(rule)
+	if currentFailureDomain == pool.FailureDomain && currentDeviceClass == pool.DeviceClass {
+		logger.Infof("pool %q has the expected failure domain %q and deviceClass %q", pool.Name, pool.FailureDomain, pool.DeviceClass)
 		return nil
 	}
-	if currentFailureDomain == "" {
-		logger.Warningf("failure domain not found for crush rule %q, proceeding to create a new crush rule", details.CrushRule)
+
+	if currentFailureDomain != pool.FailureDomain {
+		logger.Infof("failure domain not found for crush rule %q, proceeding to create a new crush rule", details.CrushRule)
```

Comment:
How about this wording?
```suggestion
		logger.Infof("creating a new crush rule for changed failure domain on crush rule %q", details.CrushRule)
```